apps/api/src/report/report.util.service.ts

```diff
@@ -946,4 +960,218 @@ export class ReportUtilService {
       }
     }
   }
+
+  /**
+   * Get the summary of durations for the report from milestone/duration table
+   *
+   * @param d table of durations
+   */
+  getDurationSummary(d: DurationTableEntry): DurationSummary {
+    const summary = { ha: d.ha } as DurationSummary;
+
+    if (d[STATUS.RECEIVED_NNAS_REPORT]) {
+      summary.NNAS = _.sum(NNAS_STAGE.map(m => d[m] || 0));
+      Object.assign(summary, _.pick(d, NNAS_STAGE));
+    }
+
+    if (d[STATUS.COMPLETED_NCAS]) {
+      summary['BCCNM & NCAS'] = _.sum(BCCNM_NCAS_STAGE.map(m => d[m] || 0));
+      Object.assign(summary, _.pick(d, BCCNM_NCAS_STAGE));
+    }
+
+    if (d[STATUS.JOB_OFFER_ACCEPTED]) {
+      const referralAck = d[STATUS.REFERRAL_ACKNOWLEDGED] ?? 0;
```

Comment:
Just a nit. Perhaps finishing the variable name to `const referralAcknowledged` would be good here. Just so I don't have to try and remember Ack is Acknowledged.

Reply:
Renamed.